Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -217,17 +217,17 @@ public class MapboxNavigationService: NSObject, NavigationService {
     /**
      Intializes a new `NavigationService`. Useful convienence initalizer for OBJ-C users, for when you just want to set up a service without customizing anything.
      
-     - parameter route: The route to follow.
-     - parameter routeIndex: The index of the route within the original `RouteController` object.
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
+     - parameter routeIndex: The index of the route within the original `RouteResponse` object.
      */
-    convenience init(route: Route, routeIndex: Int, routeOptions options: RouteOptions) {
-        self.init(route: route, routeIndex: routeIndex, routeOptions: options, directions: nil, locationSource: nil, eventsManagerType: nil)
+    convenience init(routeResponse: RouteResponse, routeIndex: Int, routeOptions options: RouteOptions) {
```

Comment:
#2589 mentions:
> A convenience initializer for MapboxNavigationService can wrap a Route in a RouteResponse, and a convenience initializer for NavigationViewController can wrap a Route in a RouteResponse and MapboxNavigationService.

Shall we add these convenience inits in this PR? Or what's the reason of not including them?

Reply:
My motivation not to add these was that all of the routes are supposed to be a result of API calling, and thus are always wrapped in `RouteResponse`. If, for some uncommon reason, user is creating a `Route` object differently, he should take care about correctly [wrapping it](https://github.com/mapbox/mapbox-directions-swift/blob/main/Sources/MapboxDirections/RouteResponse.swift#L41), providing other required fields like options and identifier.

Reply:
Hi @Udumft ,

I have a use case in which I would prefer injecting a Route and am wondering if I now need to create a convenience init to support? It seems strange to move from a strong required Route, to a model of optionals?

https://github.com/mapbox/straightaway-ios/blob/main/Straightaway/Modules/Map/ContinuousNavigationViewController.swift#L85

Given our UI/ViewModels are intentionally separate from network models, exposing a network response object seemed heavy. 

For additional context, in an upcoming feature change, we will be displaying multiple direction options. Which we will display, and begin navigation upon selection of a given route. 



cc: @irtemed88

Reply:
Hi @Udumft ,

I was seeing if I could get your suggestion for how I could approach this? I took a first stab at the upgrade migration.
https://github.com/mapbox/straightaway-ios/pull/1047